src/ucp/tag/rndv.c

```
@@ -67,6 +68,30 @@ size_t ucp_tag_rndv_pack_rkey(ucp_request_t *sreq, ucp_lane_index_t lane,
     return 0;
 }
 
+size_t ucp_tag_rndv_pack_recv_rkey(ucp_request_t *rreq, ucp_ep_h ep,
+                                   ucp_lane_index_t lane,
+                                   void *rkey_buf, uint16_t *flags)
+{
+    ucs_status_t status;
+
+    ucs_assert(UCP_DT_IS_CONTIG(rreq->recv.datatype));
+
+    /* Check if the sender needs to register the send buffer -
+     * is its datatype contiguous and does the receive side need it */
+    if (ucp_ep_rndv_md_flags(ep) & UCT_MD_FLAG_NEED_RKEY) {
+        status = ucp_request_recv_buffer_reg(rreq, ep, lane);
+        ucs_assert_always(status == UCS_OK);
```

Can we propagate the error to user level?